packages/core/src/lib/backend.ts

```diff
@@ -0,0 +1,48 @@
+import { hashToField } from './hashing'
+import { isBrowser } from 'browser-or-node'
+import type { ISuccessResult } from '../types'
+
+export interface IVerifyResponse {
+	success: boolean
+	code?: string
+	detail?: string
+	attribute?: string | null
+}
+
+export async function verifyCloudProof(
+	proof: ISuccessResult,
+	app_id: `app_${string}`,
+	action: string,
+	signal?: string,
+	endpoint?: URL | string
+): Promise<IVerifyResponse> {
+	if (isBrowser) {
+		throw new Error('verifyCloudProof can only be used in the backend.')
+	}
+
+	if (!endpoint) {
+		endpoint = new URL('https://developer.worldcoin.org/api/v2/verify')
+	}
+
+	if (typeof endpoint === 'string') {
+		endpoint = new URL(endpoint)
+	}
+
+	const response = await fetch(endpoint.href + '/' + app_id, {
```

Comment:
What's the point of doing this !endpoint. If it's to let people pass in their own we shouldn't append anything to it. We can append app_id into our own oversion

Reply:
mainly it's convenience for testing this against the staging env -- but i agree with your point that we shouldn't modify an endpoint passed in, fixed now